command/websocket.go

```diff
@@ -0,0 +1,70 @@
+package command
+
+import (
+	"bufio"
+	"bytes"
+	"errors"
+	"io"
+	"time"
+)
+
+// WsURL returns the ws URL for the running browser window.
+// Chrome will sometimes fail to print the websocket, or run for a long
+// time, without properly exiting. To avoid blocking forever in those
+// cases, give up after ten seconds.
+func WsURL(programState *ProgramState) (string, error) {
+	const wsURLReadTimeout = 10 * time.Second
```

Comment:
Make this configurable (maybe a flag). Your test should set this flag to a lower value, like 1s, so that the test doesn't take too long.